src/services/FileUploadService.ts

```diff
@@ -78,6 +83,28 @@ export class FileUploadService {
         return FileUploadModelResponse.fromModel(entry, this.baseUrl, humanReadable);
     }
 
+    public async expires(entry: FileUploadModel, expires: string): Promise<void> {
+        let value: number = ObjectUtils.getNumber(expires);
+        let timefactor: TIME_UNIT = TIME_UNIT.minutes;
+
+        if (value === 0) {
+            await this.repo.deleteEntry(entry.token);
+            throw new BadRequest(`Unable to parse expire value from ${expires}`);
+        }
+        if (expires.includes('d')) {
+            timefactor = TIME_UNIT.days;
+        } else if (expires.includes('h')) {
+            timefactor = TIME_UNIT.hours;
+        }
+        value = ObjectUtils.convertToMilli(value, timefactor);
+        if (value > entry.expiresIn) {
+            await this.repo.deleteEntry(entry.token);
+            throw new BadRequest('Cannot extend time remaining beyond original');
```

Comment:
it might be worth putting in what the calculated expiry time is. as they weill not know. maybe add it as a human readable value

Reply:
Agreed, added that information